weldx/asdf/file.py

```diff
@@ -256,22 +258,28 @@ def __init__(
             )
 
         # If we have data to write, we do it first, so a WeldxFile is always in sync.
-        if tree or new_file_created:
-            asdf_file = self._write_tree(
-                filename_or_file_like,
-                tree,
-                asdffile_kwargs,
-                write_kwargs,
-                new_file_created,
-            )
-            if isinstance(filename_or_file_like, SupportsFileReadWrite):
-                filename_or_file_like.seek(0)
-        else:
-            asdf_file = open_asdf(
-                filename_or_file_like,
-                mode=self.mode,
-                **asdffile_kwargs,
-            )
+        with warnings.catch_warnings():
+            warnings.filterwarnings(
+                category=AsdfWarning,
+                action="error",
+                message="asdf.extensions plugin from package weldx.*",
```

Comment:
this only applies to our weldx plugin right (the same warning from a different plugin would stay a warning?)

Reply:
Exactly, it just matches our extension, for other plugins it will just print warnings.